### PR TITLE
 Implement Maximum Sum of Non-Adjacent Elements

### PR DESCRIPTION
**Pull request by Guhapriya01**
This PR adds an algorithm to find the maximum sum of non-adjacent elements in an array. The implementation includes two approaches:

**Dynamic Programming with O(n) Space Complexity:**

- Utilizes a dynamic programming array to store the maximum sum at each index.
- Time Complexity: O(n)
- Space Complexity: O(n)

**Optimized Space Complexity Approach:**

- Uses two variables to track the maximum sums, reducing space complexity to O(1).
- Time Complexity: O(n)
- Space Complexity: O(1)

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

**Related Issues :** Issue #5510

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix Clang PR check and add some JUnit tests (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by Guhapriya01** [comment]
Hi @siriak , I've addressed your feedback by fixing the Clang PR check and adding JUnit tests for the `MaximumSumOfNonAdjacentElements` class.

File: src/test/java/com/thealgorithms/dynamicprogramming/MaximumSumOfNonAdjacentElementsTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Test
+    public void testGetMaxSumApproach1() {
+        // Test with various cases
+        assertEquals(15, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {3, 2, 5, 10, 7})); // 3 + 7 + 5
+        assertEquals(10, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {5, 1, 1, 5})); // 5 + 5
+        assertEquals(0, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {})); // Empty array
+        assertEquals(1, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {1})); // Single element
+        assertEquals(2, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {1, 2})); // Take max of both
+        assertEquals(3, MaximumSumOfNonAdjacentElements.getMaxSumApproach1(new int[] {3, 2})); // Take 3
```
Comment (siriak) [inline src/test/java/com/thealgorithms/dynamicprogramming/MaximumSumOfNonAdjacentElementsTest.java:17]: Please create parameterized tests or separate test cases instead

Comment (Guhapriya01) [reply src/test/java/com/thealgorithms/dynamicprogramming/MaximumSumOfNonAdjacentElementsTest.java:17]: I'll refactor the tests using parameterized tests or separate cases. I'll update the pull request shortly.

**Review by siriak — COMMENTED** [review]
Please check why build PR check is failing

**Review by siriak — APPROVED** [review]

**Pull request merged**